File: Serenity.Script.UI/Grid/QuickFilter.cs
using System;
using System.Collections;
using System.Collections.Generic;
using jQueryApi;
using System.Linq;
using System.Runtime.CompilerServices;

namespace Serenity
{
    [Imported, Serializable, IncludeGenericArguments(false)]
    public class QuickFilter<TWidget, TOptions>
        where TWidget: Widget
    {
        public string Field { get; set; }
        public string Title { get; set; }
        public Type Type { get; set; }
        public TOptions Options { get; set; }
        public Action<jQueryObject> Element { get; set; }
        public Action<TWidget> Init { get; set; }
        public Action<QuickFilterArgs<TWidget>> Handler { get; set; }
        public bool Seperator { get; set; }
        public string CssClass { get; set; }
    }
}

File: src/Serenity.Scripts/CoreLib/UI/DataGrid/QuickFilter.ts
import type { Widget } from "../Widgets/Widget";

export interface QuickFilterArgs<TWidget> {
    field?: string;
    widget?: TWidget;
    request?: Serenity.ListRequest;
    equalityFilter?: any;
    value?: any;
    active?: boolean;
    handled?: boolean;
}

export interface QuickFilter<TWidget extends Widget<TOptions>, TOptions> {
    field?: string;
    type?: new (element: JQuery, options: TOptions) => TWidget;
    handler?: (h: QuickFilterArgs<TWidget>) => void;
    title?: string;
    options?: TOptions;
    element?: (e: JQuery) => void;
    init?: (w: TWidget) => void;
    separator?: boolean;
    cssClass?: string;
    loadState?: (w: TWidget, state: any) => void;
    saveState?: (w: TWidget) => any;
    displayText?: (w: TWidget, label: string) => string;
}